test/clj/game_test/cards/icebreakers.clj

```diff
@@ -272,45 +272,43 @@
       (run-continue state)
       (is (find-card "Faerie" (:discard (get-runner))) "Faerie trashed"))))
 
-(deftest faust-pump
-  ;; Faust - Pump by discarding
-  (do-game
-    (new-game (default-corp)
-              (default-runner [(qty "Faust" 1) (qty "Sure Gamble" 3)]))
-    (take-credits state :corp)
-    (play-from-hand state :runner "Faust")
-    (let [faust (get-in @state [:runner :rig :program 0])]
-      (card-ability state :runner faust 1)
-      (prompt-card :runner (first (:hand (get-runner))))
-      (is (= 4 (:current-strength (refresh faust))) "4 current strength")
-      (is (= 1 (count (:discard (get-runner)))) "1 card trashed"))))
-
-(deftest faust-pump
-  ;; Faust - Pump does not trigger trash prevention. #760
-  (do-game
-    (new-game (default-corp)
-              (default-runner [(qty "Faust" 1)
-                               (qty "Sacrificial Construct" 1)
-                               (qty "Fall Guy" 1)
-                               (qty "Astrolabe" 1)
-                               (qty "Gordian Blade" 1 )
-                               (qty "Armitage Codebusting" 1)]))
-    (take-credits state :corp)
-    (core/draw state :runner 1)
-    (play-from-hand state :runner "Faust")
-    (play-from-hand state :runner "Fall Guy")
-    (play-from-hand state :runner "Sacrificial Construct")
-    (is (= 2 (count (get-in @state [:runner :rig :resource]))) "Resources installed")
-    (let [faust (get-in @state [:runner :rig :program 0])]
-      (card-ability state :runner faust 1)
-      (prompt-card :runner (find-card "Astrolabe" (:hand (get-runner))))
-      (is (empty? (:prompt (get-runner))) "No trash-prevention prompt for hardware")
-      (card-ability state :runner faust 1)
-      (prompt-card :runner (find-card "Gordian Blade" (:hand (get-runner))))
-      (is (empty? (:prompt (get-runner))) "No trash-prevention prompt for program")
-      (card-ability state :runner faust 1)
-      (prompt-card :runner (find-card "Armitage Codebusting" (:hand (get-runner))))
-      (is (empty? (:prompt (get-runner))) "No trash-prevention prompt for resource"))))
+(deftest faust
+  (testing "Basic test: Pump by discarding"
```

Comment:
Does this mean one cannot use `lein test :only` to test just the "basic test" here?

Reply:
To test only the `testing` section of this test? Yeah, that's what these changes do. I've been refactoring the tests like this so that there's a one-to-one relationship between the cards and the test functions. The equivalent would be a single test class with multiple methods for a given class in Java.

I feel this is a positive for development, because it means you're testing _all_ functionality against a card when making changes instead of having to remember to test each interaction individually or running the whole card type test suite.

Reply:
Sounds good. I like the reasoning.